### PR TITLE
Add central publish connector template workflow and Add trivy scan step to release connector workflow

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose
- Related issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/4778

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by ThisaruGuruge — DISMISSED** [review]

File: .github/workflows/central-publish-connector-template.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        env:
+          BALLERINA_DEV_CENTRAL: false
+          BALLERINA_STAGE_CENTRAL: true
+          BALLERINA_CENTRAL_ACCESS_TOKEN: ${{ secrets.BALLERINA_CENTRAL_STAGE_ACCESS_TOKEN }}
+          packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
+          packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
+        run: |
+          sed -i 's/version=\(.*\)-SNAPSHOT/version=\1/g' gradle.properties
+          ./gradlew clean build -PpublishToCentral=true ${{ inputs.additional-publish-flags }}
```
Comment (NipunaRanasinghe) [inline .github/workflows/central-publish-connector-template.yml:64]: Don't we need to have a step to publish to the central prod env as well?

Comment (ThisaruGuruge) [reply .github/workflows/central-publish-connector-template.yml:64]: No, that is handled by the release workflow. This is specifically for the DEV and STAGE publishes.

Comment (NipunaRanasinghe) [reply .github/workflows/central-publish-connector-template.yml:64]: @ThisaruGuruge Ack. if thats the case, shall we change the workflow name and the description accordingly?

Comment (MohamedSabthar) [reply .github/workflows/central-publish-connector-template.yml]: updated

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**